tests/UnixServerTest.php

```diff
@@ -0,0 +1,277 @@
+        fwrite($client, "foo\n");
+        stream_socket_shutdown($client, STREAM_SHUT_WR);
+
+        $mock = $this->expectCallableOnceWith("foo\n");
+
+        $this->server->on('connection', function ($conn) use ($mock) {
+            $conn->on('data', $mock);
+        });
+        $this->loop->tick();
+        $this->loop->tick();
+    }
+
+    public function testLoopWillEndWhenServerIsClosed()
+    {
+        // explicitly unset server because we already call close()
+        $this->server->close();
+        $this->server = null;
+
+        $this->loop->run();
+    }
+
+    public function testCloseTwiceIsNoOp()
+    {
+        $this->server->close();
+        $this->server->close();
+    }
+
+    public function testGetAddressAfterCloseReturnsNull()
+    {
+        $this->server->close();
+        $this->assertNull($this->server->getAddress());
+    }
+
+    public function testLoopWillEndWhenServerIsClosedAfterSingleConnection()
+    {
+        $client = stream_socket_client($this->uds);
+
+        // explicitly unset server because we only accept a single connection
+        // and then already call close()
+        $server = $this->server;
+        $this->server = null;
+
+        $server->on('connection', function ($conn) use ($server) {
+            $conn->close();
+            $server->close();
+        });
+
+        $this->loop->run();
+    }
+
+    public function testDataWillBeEmittedInMultipleChunksWhenClientSendsExcessiveAmounts()
+    {
+        $client = stream_socket_client($this->uds);
+        $stream = new DuplexResourceStream($client, $this->loop);
+
+        $bytes = 1024 * 1024;
+        $stream->end(str_repeat('*', $bytes));
+
+        $mock = $this->expectCallableOnce();
+
+        // explicitly unset server because we only accept a single connection
+        // and then already call close()
+        $server = $this->server;
+        $this->server = null;
+
+        $received = 0;
+        $server->on('connection', function ($conn) use ($mock, &$received, $server) {
+            // count number of bytes received
+            $conn->on('data', function ($data) use (&$received) {
+                $received += strlen($data);
+            });
+
+            $conn->on('end', $mock);
+
+            // do not await any further connections in order to let the loop terminate
+            $server->close();
+        });
+
+        $this->loop->run();
+
+        $this->assertEquals($bytes, $received);
+    }
+
+    /**
+     * @covers React\EventLoop\StreamSelectLoop::tick
+     */
+    public function testConnectionDoesNotEndWhenClientDoesNotClose()
+    {
+        $client = stream_socket_client($this->uds);
+
+        $mock = $this->expectCallableNever();
+
+        $this->server->on('connection', function ($conn) use ($mock) {
+            $conn->on('end', $mock);
+        });
+        $this->loop->tick();
+        $this->loop->tick();
+    }
+
+    /**
+     * @covers React\EventLoop\StreamSelectLoop::tick
```

Comment:
Why does this test covers `React\EventLoop\StreamSelectLoop::tick`?

Reply:
I must admit that the tests are not original art but all copied from `TcpServer`- so that's a copy/paste artefact. Remove the covers hint here?

Reply:
Yeah noticed that, but to be honest that is alright. There are few distinct differences that I'm not bothered by it 😄 . The hints are fine as long as they make sense, covering a loop tick in a server test doesn't =D

Reply:
Should I change the TcpServerTest, too?

Reply:
Nah out of scope for this PR